### DISCUSSION TITLE
OnlyDirty with required fields to log

### DISCUSSION THREAD

**Discussion by vmiguellima** [spatie/laravel-activitylog#748] [Q&A] [answered]
Hi, after a lot research I couldn't find a way to so it, so I'm here if I can get some help.

Problem Structure:

I have an Article model where I only want to log field that were changed, so I use ```protected static $logOnlyDirty = true;```.
And i want to show a chronological order of the events, like: "Title Updated".  "Slug Updated".

But using the ```logOnlyDirty```, will only log the changes. Being that, how i will establishe a relationship between logs and and the Article?

My suggestion is like:

```protected static $logRequired = ["id", "slug"];```

I love your packages,
Miguel

**Comment by Gummibeer** [comment, marked as answer]
Hey,
the `Activity` model has a `subject` relationship which always points to the subject model of this activity.
For model event activity logs this relationship is always the model that triggers the activity.
For custom activities you can set this relationship your own.
https://docs.spatie.be/laravel-activitylog/v3/advanced-usage/logging-model-events/

**Reply by vmiguellima** [reply]
Thank you very much for the help!
